[Magic] Convert Enhancing spells

Xaver-DaRed opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Convert White, Black and Ninjutsu spells to use a global file.
Removes global files used exclusively for Bar-spells

- Spells not existing yet:
Animus- (2), Adoquium, - storm II (8), Temper III

- New spell scripts/parameters implemented:
Refresh III, Regen V

- Other notes:
Base implementation of Embolden Effect. Doesnt account for merits/JP whatever.
Auspice is treated as an en-spell. That should be checked.
To review, focus on spell_enhancing.lua, spell_enhancing_ninjutsu.lua and spell_enhancing_teleport.lua
Other file changes were mostly cleanup.
Sanity fail due to magic.lua

Xaver-DaRed committed:
Convert Protect, Shell and Regen families.

Implement Regen V

Xaver-DaRed committed:
Convert Refresh family

Implement Refresh III

Xaver-DaRed committed:
Convert Aquaveil and Bar-Element Spells

Xaver-DaRed committed:
Convert Bar-Status Spells

Xaver-DaRed committed:
Convert Blink, Boost-Stat and Gain-Stat families

Xaver-DaRed committed:
Merge branch 'LandSandBoat:base' into Magic/Enhancing

Xaver-DaRed committed:
Convert Teleport-, Recall-, Warp-, Retrace and Escape Spells

Xaver-DaRed committed:
Merge remote-tracking branch 'origin/Magic/Enhancing' into Magic/Enhancing

Xaver-DaRed committed:
Merge branch 'base' into Magic/Enhancing

Xaver-DaRed committed:
Fix lua errors

Xaver-DaRed committed:
Merge remote-tracking branch 'origin/Magic/Enhancing' into Magic/Enhancing

Xaver-DaRed committed:
Convert Deodorize, Invis, Sneak, Haste and Embrava

Xaver-DaRed committed:
Conver En-Spells, Phalanx and Spikes

Xaver-DaRed committed:
Convert Flurry, Temper and -storms

Xaver-DaRed committed:
Convert Ninjutsu buffs. Change song global file name. Cleanup.

Xaver-DaRed committed:
Convert Auspice, Crusade and Foil

Xaver-DaRed committed:
Merge remote-tracking branch 'origin/base' into Magic/Enhancing

claywar commented on scripts/globals/spells/black/retrace.lua:
```diff
@@ -2,28 +2,16 @@
-require("scripts/globals/teleports")
-require("scripts/globals/status")
-require("scripts/globals/msg")
+require("scripts/globals/spells/spell_enhancing_teleport")
```
If there's multiple enhancing scripts in the global, perhaps we should move this into a sub-directory?  Just thinking about really long names.

claywar commented on scripts/globals/spells/songs/fowl_aubade.lua:
```diff
@@ -11,7 +11,7 @@ spell_object.onMagicCastingCheck = function(caster, target, spell)
 
 spell_object.onSpellCast = function(caster, target, spell)
-    return xi.spells.spell_song_enhancing.useEnhancingSong(caster, target, spell)
+    return xi.spells.spell_enhancing_song.useEnhancingSong(caster, target, spell)
```
Similar to the naming convention idea for sub-directories.  The global table and function name seem to be a bit redundant.  Not sure if we need to be as verbose (for example, the table could be xi.spells.enhancing, and the function wouldn't really need to have enhancing in its name as well.

claywar commented on scripts/globals/spells/spell_enhancing.lua:
```diff
@@ -0,0 +1,344 @@
+-- 1 Main function, called by spell scripts.
+
+-- Table variables.
+local enhancingTable = xi.spells.parameters.enhancingSpell
```
I would almost prefer having parameters used in this script here for easier reference instead of one massive file containing params for all different spell types.

claywar commented on scripts/globals/spells/spell_enhancing.lua:
```diff
@@ -0,0 +1,344 @@
+        duration = duration * targetLevel / spellLevel
+    end
+
+    ----------
```
Bit of an unnecessary comment at the end of this function

Xaver-DaRed replied:
Removed

TeoTwawki reviewed (commented):
I think we could reduce the if/else-yness of the global code with some tabled function calling per discord discussion - worth a looking into at least

WinterSolstice8 commented:
> I think we could reduce the if/else-yness of the global code with some tabled function calling per discord discussion - worth a looking into at least

per discussion we could mark it TODO in the code to get this merged

Xaver-DaRed committed:
Tested. Added TODOs

As spoken with Claywar, next step is to simplify function names across all converted spells

Xaver-DaRed commented:
I spoke with Claywar and his comments will be adressed in a different PR. One that doesn't involve any new coding, just simplifying function names and getting rid of the global parameters.lua file

Xaver-DaRed committed:
Merge branch 'LandSandBoat:base' into Magic/Enhancing

Xaver-DaRed committed:
Merge branch 'base' into Magic/Enhancing

Xaver-DaRed commented:
Ready.
CI fail is because magic.lua file (cyclomatic complexity).
So: not my fault^^
